### PR TITLE
CFE-3562 Changed custom promise type interpreter attribute to be optional

### PR DESCRIPTION
**Pull request by craigcomstock**
If interpreter provided with no path attribute then error.

Ticket: CFE-3562
Changelog: title

### CODE REVIEW

File: tests/acceptance/30_custom_promise_types/13_binary_path.cf [hunk 1]
```diff
@@ -0,0 +1,21 @@
+promise agent binary
+{
+  path => "/tmp/binary";
+}
+
+bundle agent main
+{
+  commands:
+    "/usr/bin/gcc $(this.promise_dirname)/binary.c -o /tmp/binary"
```
Comment (craigcomstock) [inline tests/acceptance/30_custom_promise_types/13_binary_path.cf:9]: `/usr/bin/gcc` too much of an assumption? ;) Will find out.

Comment (craigcomstock) [reply tests/acceptance/30_custom_promise_types/13_binary_path.cf:9]: Yes. Fixed.

File: tests/acceptance/30_custom_promise_types/13_binary_path.cf [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+promise agent binary
+{
+  path => "$(G.cwd)/custom_promise_binary";
+}
+
+bundle agent test
+{
+  binary:
+    "foobar"
+    classes => if_ok("binary_ok");
```
Comment (olehermanse) [inline tests/acceptance/30_custom_promise_types/13_binary_path.cf:17]:
```suggestion
      classes => if_ok("binary_ok");
```

File: tests/acceptance/30_custom_promise_types/custom_promise_binary.c [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    printf("result=valid\n");
+    printf("\n");
+
+    if ( getline(&line, &len, stdin) == -1 )
+    {
+      perror( "Couldn't read from stdin" );
+      return 1;
+    }
+
+    printf("operation=evaluate_promise\n");
```
Comment (olehermanse) [inline tests/acceptance/30_custom_promise_types/custom_promise_binary.c:28]: No big deal, but spec says that repaired should have an info log message. I would change it to `kept` or add a log message:

```suggestion
    printf("log_info=The foobar was fixed\n");
    printf("operation=evaluate_promise\n");
```

**Review by olehermanse — APPROVED** [review]
Looks good to me, thanks!

File: libpromises/mod_custom.c [hunk 4]
```diff
@@ -411,17 +411,24 @@ static void PromiseModule_DestroyInternal(PromiseModule *module)
 
 static PromiseModule *PromiseModule_Start(char *interpreter, char *path)
 {
-    assert(interpreter != NULL);
     assert(path != NULL);
 
-    if (access(interpreter, X_OK) != 0)
+    if ( (interpreter != NULL) && (access(interpreter, X_OK) != 0) )
```
Comment (vpodzime) [inline libpromises/mod_custom.c:416]: Please remove the spaces inside parentheses. We don't use this pattern, AFAICT. Applies to the other `if`s below as well.

**Review by vpodzime — COMMENTED** [review]
Looks good to me otherwise.

**Review by vpodzime — APPROVED** [review]
ACK

**Pull request merged**